lib/bots/syncbot.ts

```diff
@@ -295,11 +301,18 @@ export class SyncBot extends ProcBot {
      * @param error  Error to report.
      * @param event  Event that caused the error.
      */
-    private logError(error: Error, event: MessengerContext): void {
+    private logError(error: Error, event: InterimContext): void {
         // Do what we can to make this event obvious in the logs
         this.logger.log(LogLevel.WARN, 'v!!!v');
         this.logger.log(LogLevel.WARN, error.message);
-        this.logger.log(LogLevel.WARN, JSON.stringify(event));
+        const logEvent = _.cloneDeep(event);
+        if (logEvent.toIds.token) {
+            logEvent.toIds.token = 'redacted';
+        }
+        if (logEvent.sourceIds.token) {
+            logEvent.sourceIds.token = 'redacted';
+        }
+        this.logger.log(LogLevel.WARN, JSON.stringify(logEvent));
```

Comment:
Can we not make the logger itself redact the sensitive tokens with a `str.replace(new RegExp(_.escapeRegExp(token), 'g'), 'redacted')`?